osu.Game/Overlays/Mods/ModCustomisationPanel.cs

```diff
@@ -214,15 +215,23 @@ public FocusGrabbingContainer(ModCustomisationPanel panel)
                 this.panel = panel;
             }
 
-            protected override void OnHoverLost(HoverLostEvent e)
+            private InputManager? inputManager;
+
+            protected override void LoadComplete()
+            {
+                base.LoadComplete();
+                inputManager = GetContainingInputManager();
+            }
+
+            protected override void Update()
             {
-                if (ExpandedState.Value is ModCustomisationPanelState.ExpandedByHover
-                    && !ReceivePositionalInputAt(e.ScreenSpaceMousePosition))
+                base.Update();
+
+                if (ExpandedState.Value == ModCustomisationPanelState.ExpandedByHover)
                 {
-                    ExpandedState.Value = ModCustomisationPanelState.Collapsed;
+                    if (!ReceivePositionalInputAt(inputManager!.CurrentState.Mouse.Position) && inputManager.DraggedDrawable == null)
```

Comment:
Bit baffled by this NRT usage, why declare `inputManager` as `InputManager?` if you're just going to silence null inspections at point of usage here? Just make the field non-nullable from the start?

Also is there any point to splitting this into two nested `if`s rather than adding a third condition?

Reply:
Reads better to me, not a big fan of very long conditionals. Probably just me.

Reply:
Silence at point of assignment IMO.